### PR TITLE
Minor improvements to CRefine

### PR DESCRIPTION
**Pull request by corlewis**

### CODE REVIEW

**Review by lsf37 — APPROVED** [review]
Nice. As discussed out-of-band, we should leave this one queued up for a bit until we can integrate nicely with AArch64 and MCS.

File: proof/crefine/ARM/TcbAcc_C.thy
Comment (michaelmcinerney) [inline proof/crefine/ARM/TcbAcc_C.thy:54]: Another spot where we should be able to remove `UNIV`

**Review by michaelmcinerney — APPROVED** [review]
👍

File: proof/crefine/ARM/SR_lemmas_C.thy
Comment (corlewis) [reply proof/crefine/ARM/SR_lemmas_C.thy:908]: That could work, although there isn't anything directly comparable there after the `install_C_file` command.

The other option I just thought of is `cspec/KernelState_C`. That also doesn't have anything directly comparable, but has the benefit of being arch-generic so there's less duplication.

**Comment by corlewis** [comment]
This is now updated to also include AArch64. I think it's now ready to merge, pending a decision on the location of `ThreadState_defs`.

File: proof/crefine/ARM/SR_lemmas_C.thy
Comment (Xaphiosis) [reply proof/crefine/ARM/SR_lemmas_C.thy:908]: I don't know how directly comparable you need it to be, but from my perspective the `Hide unqualified names conflicting with Kernel_Config names. Force use of Kernel_C prefix` sounds like it's along the lines of what you're trying to do. It does have the downside of being duplicated across arches, being in `Kernel_C`. I've never touched `KernelState_C` before, it looks ancient. @lsf37 might remember what its role is supposed to be. I'd still prefer it in `Kernel_C` for now, given it's somewhere we might find it again.

Comment (corlewis) [reply proof/crefine/ARM/SR_lemmas_C.thy:908]: That's a good point about being able to find this again. I'm not sure if I've seen `KernelState_C` before either, I only found it now because I was looking at the arch-generic `cspec` files to decide whether any were suitable.

Comment (Xaphiosis) [reply proof/crefine/ARM/SR_lemmas_C.thy:908]: It's a solid idea, I just don't know what that file was meant for. Helpfully it lacks a comment at the top, maintaining an air of mystery. No objection to shoving it in there though; you're adding a lemma, not removing/hiding stuff, so `Kernel_C` isn't forced. We'll see what democracy says.

Comment (lsf37) [reply proof/crefine/ARM/SR_lemmas_C.thy:908]: `KernelState_C` was introduced by Simon in 2008 with the commit message "Changes for new bitfields", so I guess we can take it as whatever we like :-), but it looks like it is not actually included anywhere and the definitions in there are unused.

I like the idea of a theory where we could put some generic things that work for all kernel versions, but introducing that will mess with the theory include graph a bit. Not a major problem, but might be better to defer to a point where we want to think about arch splitting CRefine a bit more.

So I propose to put `ThreadState_defs` into `Kernel_C` instead and to remove `KernelState_C` from `ROOTS` and the repo.

Comment (lsf37) [reply proof/crefine/ARM/SR_lemmas_C.thy:908]: Hm, correction. As Corey points out: the file *is* used, from generated bitfield theories (which `git grep` doesn't see), and those do use the definitions. In that case we should not remove it, of course.

Ok, new proposal: I'd still add `ThreadState_defs` to `Kernel_C`, and add a comment to `KernelState_C` saying that it is the base theory for generated bitfield proofs about the kernel.

Comment (corlewis) [reply proof/crefine/ARM/SR_lemmas_C.thy]: I like the proposal as well and have implemented it. Is there anything else to do before merging this PR?

Comment (lsf37) [reply proof/crefine/ARM/SR_lemmas_C.thy]: All good to go from my side.

**Review by Xaphiosis — APPROVED** [review]
I think this is fine now

**Pull request merged**
